lang/src/org/partiql/lang/eval/io/DelimitedValues.kt

```diff
@@ -66,38 +68,29 @@ object DelimitedValues {
     @JvmStatic
     fun exprValue(valueFactory: ExprValueFactory,
                   input: Reader,
-                  delimiter: String,
+                  delimiter: Char,
                   hasHeader: Boolean,
                   conversionMode: ConversionMode): ExprValue {
         val reader = BufferedReader(input)
-        val columns: List<String> = when {
-            hasHeader -> {
-                val line = reader.readLine()
-                    ?: throw IllegalArgumentException("Got EOF for header row")
-
-                line.split(delimiter)
-            }
-            else -> emptyList()
+        val csvFormat = when (hasHeader){
+            true -> CSVFormat.DEFAULT.withDelimiter(delimiter).withTrim().withFirstRecordAsHeader()
+            false -> CSVFormat.DEFAULT.withDelimiter(delimiter).withTrim()
         }
+        val csvParser = CSVParser(reader, csvFormat)
+        val columns: List<String> = csvParser.headerNames // `columns` is an empty list when `hasHeader` is false
 
-        val seq = generateSequence {
-            val line = reader.readLine()
-            when (line) {
-                null -> null
-                else -> {
-                    valueFactory.newStruct(
-                        line.splitToSequence(delimiter).mapIndexed {i , raw ->
-                            val name = when {
-                                i < columns.size -> columns[i]
-                                else -> syntheticColumnName(i)
-                            }
-                            conversionMode.convert(valueFactory, raw).namedValue(valueFactory.newString(name))
-                        },
-                        StructOrdering.ORDERED
-                    )
-                }
-            }
-        }
+        val seq = csvParser.asSequence().map { csvRecord ->
+            valueFactory.newStruct(
+                csvRecord.mapIndexed { i, value ->
+                    val name = when {
+                        i < columns.size -> columns[i]
+                        else -> syntheticColumnName(i)
+                    }
```

Comment:
The code here allows the case that each row has different numbers of fields. I just follow the original code even though [rfc4180](https://datatracker.ietf.org/doc/html/rfc4180) does not allow it. Let me know if it is better to change it.